pnet_macros/src/decorator.rs

```diff
@@ -211,6 +211,10 @@ fn make_packet(ecx: &mut ExtCtxt, span: Span, name: String, sd: &ast::StructDef)
 }
 
 fn make_packets(ecx: &mut ExtCtxt, span: Span, item: &ast::Item) -> Option<Vec<Packet>> {
+    if item.vis != ast::Visibility::Public {
+        ecx.span_err(item.span, "#[packet] structs must be public");
```

Comment:
I suppose you'll also need to update the test case with the relevant error message too.